Tests/test_image.py

```diff
@@ -315,6 +315,32 @@ def test_offset_not_implemented(self):
     def test_fromstring(self):
         self.assertRaises(NotImplementedError, Image.fromstring)
 
+    def test_linear_gradient_wrong_mode(self):
+        # Arrange
+        wrong_mode = "RGB"
+
+        # Act / Assert
+        self.assertRaises(ValueError,
+                          lambda: Image.linear_gradient(wrong_mode))
+        return
+
+    def test_linear_gradient(self):
+
+        # Arrange
+        for mode in ["L", "P"]:
+
+            # Act
+            im = Image.linear_gradient(mode)
+
+            # Assert
+            self.assertEqual(im.size, (256, 256))
+            self.assertEqual(im.mode, mode)
+            self.assertEqual(im.getpixel((0, 0)), 0)
+            self.assertEqual(im.getpixel((255, 255)), 255)
+            im2 = Image.open('Tests/images/linear_gradient_{}.png'.format(
```

Comment:
What I meant is that we only need one target image, not that we only need to test one mode. 

e.g. `target = Image.open('Tests/images/linear_gradient.png').convert(mode)`